ksp/src/main/kotlin/com/tschuchort/compiletesting/Ksp.kt

```diff
@@ -23,11 +23,11 @@ import java.io.File
  * The list of symbol processors for the kotlin compilation.
  * https://goo.gle/ksp
  */
-var KotlinCompilation.symbolProcessors: List<SymbolProcessor>
-    get() = getKspRegistrar().processors
+var KotlinCompilation.symbolProcessorProviders: List<SymbolProcessorProvider>
+    get() = getKspRegistrar().providers
```

Comment:
Am I correct to assume that this is also a breaking change in KSP itself? If so, then I guess it's acceptable for us to break the API here, otherwise we should probably offer some kind of migration path.

Reply:
Yes, it does have support to load the old class at least for now https://github.com/google/ksp/commit/7dcacc6510bc7a3bc21f873424db09ad0237086c . The being said, personally I'd be updating kotlin-compile-testing in order to do it the new way so I don't mind the breaking change.

Reply:
btw, just updating KSP already broke this in Room, not sure why, investigating.

So we are probably already broken w/ latest KSP even though it is backwards compatible on the KSP side (this is "probably") because we subclass KSP's registrar.

```
RAW OUTPUT:
    e: java.lang.AbstractMethodError: Receiver class com.tschuchort.compiletesting.KspTestExtension does not define or inherit an implementation of the resolved method 'abstract java.util.List loadProviders()' of abstract class com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension.
        at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension.doAnalysis(KotlinSymbolProcessingExtension.kt:138)
        at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration(TopDownAnalyzerFacadeForJVM.kt:116)
        at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration$default(TopDownAnalyzerFacadeForJVM.kt:85)
        at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:517)
        at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:508)
        at org.jetbrains.kotlin.cli.common.messages.AnalyzerWithCompilerReport.analyzeAndReport(AnalyzerWithCompilerReport.kt:114)

```

Reply:
I suppose we can live with not maintaining the old interface then.